### PR TITLE
feat: support PolarGridRPhi2 segmentation in CalorimeterHitReco

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This supports PolarGridRPhi2 segmentations in CalorimeterHitReco, similar to the CartesianGridXY. [As far as I know](https://github.com/search?q=repo%3Aeic%2Fepic%20PolarGridRPhi2&type=code), only used by HcalEndcapN so impact needs to be assessed by that group.

### What kind of change does this PR introduce?
- [x] Bug fix (issue #700)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
Maybe, cell dimension not written as bounding box but as actual cell dimensions, may affect clustering downstream.

### Does this PR change default behavior?
Yes, cell dimension not written as bounding box but as actual cell dimensions, may affect clustering downstream.

### CODE REVIEW

File: src/algorithms/calorimetry/CalorimeterHitReco.cc [hunk 1]
```diff
@@ -215,9 +215,14 @@ void CalorimeterHitReco::AlgorithmProcess() {
...
             m_log->debug("Using segmentation for cell dimensions: {}", fmt::join(cdim, ", "));
+        } else if (segmentation_type == "PolarGridRPhi2") {
+            cdim.resize(3);
+            cdim[0] = cell_dim[0];
+            cdim[1] = cell_dim[1];
+            m_log->debug("Using segmentation for cell dimensions: {}", fmt::join(cdim, ", "));
         } else {
             if (segmentation_type != "NoSegmentation") {
-                m_log->warn("Usupported segmentation type \"{}\"", segmentation_type);
+                m_log->warn("Unsupported segmentation type \"{}\"", segmentation_type);
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterHitReco.cc:225]: Gratuitous typo correction.

File: src/algorithms/calorimetry/CalorimeterHitReco.cc [hunk 2]
```diff
@@ -215,9 +215,14 @@ void CalorimeterHitReco::AlgorithmProcess() {
             cdim[0] = cell_dim[0];
             cdim[1] = cell_dim[1];
             m_log->debug("Using segmentation for cell dimensions: {}", fmt::join(cdim, ", "));
+        } else if (segmentation_type == "PolarGridRPhi2") {
+            cdim.resize(3);
+            cdim[0] = cell_dim[0];
+            cdim[1] = cell_dim[1];
```
Comment (veprbl) [inline src/algorithms/calorimetry/CalorimeterHitReco.cc:221]: Makes no sense to implement this one. Downstream code expects things to be comparable in XYZ with cm as units.

Comment (wdconinc) [reply src/algorithms/calorimetry/CalorimeterHitReco.cc:221]: So is bounding box what we want? We should at least be explicit about it.

Comment (veprbl) [reply src/algorithms/calorimetry/CalorimeterHitReco.cc:221]: No, we don't want it either, and likely we simply don't use the `dimensions` field for EndcapNHcal, at least in clusterer. This design is broken, I will be pushing to drop this eventually.

Comment (wdconinc) [reply src/algorithms/calorimetry/CalorimeterHitReco.cc:221]: But we should do something that doesn't cause an error message for every event. It's affecting ops.

Comment (veprbl) [reply src/algorithms/calorimetry/CalorimeterHitReco.cc:221]: Didn't we hide the message into debug?

Comment (wdconinc) [reply src/algorithms/calorimetry/CalorimeterHitReco.cc:221]: Not the message for when the segmentation isn't recognized. That's still a warning, as it probably should be.

**Pull request closed without merging**